web/core/src/jsMain/kotlin/androidx/compose/web/css/CSSUnits.kt

```diff
@@ -1,114 +1,270 @@
 package org.jetbrains.compose.web.css
 
-interface CSSSizeValue : CSSUnitValue, StylePropertyValue
-
-interface CSSRelValue : CSSSizeValue
-interface CSSpercentValue : CSSRelValue
-interface CSSemValue : CSSRelValue
-interface CSSexValue : CSSRelValue
-interface CSSchValue : CSSRelValue
-interface CSSicValue : CSSRelValue
-interface CSSremValue : CSSRelValue
-interface CSSlhValue : CSSRelValue
-interface CSSrlhValue : CSSRelValue
-interface CSSvwValue : CSSRelValue
-interface CSSvhValue : CSSRelValue
-interface CSSviValue : CSSRelValue
-interface CSSvbValue : CSSRelValue
-interface CSSvminValue : CSSRelValue
-interface CSSvmaxValue : CSSRelValue
-interface CSScmValue : CSSRelValue
-interface CSSmmValue : CSSRelValue
-interface CSSQValue : CSSRelValue
-
-interface CSSAbsValue : CSSSizeValue
-interface CSSptValue : CSSAbsValue
-interface CSSpcValue : CSSAbsValue
-interface CSSpxValue : CSSAbsValue
-
-interface CSSangleValue : CSSUnitValue
-interface CSSdegValue : CSSangleValue
-interface CSSgradValue : CSSangleValue
-interface CSSradValue : CSSangleValue
-interface CSSturnValue : CSSangleValue
-
-interface CSSTimeValue : CSSUnitValue
-interface CSSsValue : CSSTimeValue
-interface CSSmsValue : CSSTimeValue
-
-interface CSSFrequencyValue : CSSUnitValue
-interface CSSHzValue : CSSFrequencyValue
-interface CSSkHzValue : CSSFrequencyValue
-
-interface CSSResolutionValue : CSSUnitValue
-interface CSSdpiValue : CSSResolutionValue
-interface CSSdpcmValue : CSSResolutionValue
-interface CSSdppxValue : CSSResolutionValue
-
-interface CSSFlexValue : CSSUnitValue
-interface CSSfrValue : CSSFlexValue
+interface CSSUnitValue<out T : CSSUnit> : CSSNumericValue {
+    val value: Float
+    val unit: T
+    fun asString(): String = "${value}${unit.value}"
+}
+
+private class CSSUnitValueTyped<out T : CSSUnit>(
+    override val value: Float,
+    override val unit: T
+) : CSSUnitValue<T>
+
+typealias CSSSizeValue = CSSUnitValue<CSSUnit>
+typealias CSSpxValue = CSSUnitValue<CSSUnit.px>
+
+interface CSSUnitRel : CSSUnit
+interface CSSUnitAbs: CSSUnit
+interface CSSUnitAngle: CSSUnit
+interface CSSUnitTime: CSSUnit
+interface CSSUnitFrequency: CSSUnit
+interface CSSUnitResolution: CSSUnit
+interface CSSUnitFlex: CSSUnit
+
+typealias CSSAngleValue = CSSUnitValue<CSSUnitAngle>
+
+sealed interface CSSUnit {
+    val value: String
+
+    object percent: CSSUnitRel {
+        override val value: String
```

Comment:
is there a reason to declare a getter?

can't it be just like below snippet?

```
override val value: String = "%
```

Reply:
no actual reason! ) it's remnant of previous approach, so I've pushed a simplification.